server/src/infra/database.config.ts

```diff
@@ -23,5 +23,33 @@ export const databaseConfig: PostgresConnectionOptions = {
   ...urlOrParts,
 };
 
+export const databaseConfigVector: PostgresConnectionOptions = {
+  ...databaseConfig,
+  migrationsRun: false,
+  migrations: [__dirname + '/migrations/vector/*.{js,ts}'],
+};
+
+export async function initDataSource(): Promise<DataSource> {
+  const dataSource = await new DataSource(databaseConfig).initialize();
+  
+  const hasVectorExtension = (await dataSource.query(
+    `SELECT * FROM pg_available_extensions WHERE name = 'vectors'`,
+  )).length > 0;
+
+  if (hasVectorExtension) {
+    const dataSourceVector = await new DataSource(databaseConfigVector).initialize();
+    await dataSourceVector.runMigrations();
+
+    await dataSourceVector.query(`SET vectors.enable_prefilter = on`);
+
+    await dataSourceVector.destroy();
+  }
+
+  return dataSource;
+}
+
 // this export is used by TypeORM commands in package.json#scripts
-export const dataSource = new DataSource(databaseConfig);
+export let dataSource: DataSource;
+(async () => {
+  dataSource = await initDataSource();
+})();
```

Comment:
Yeah this won't work for the e2e I think

Reply:
I reverted this change for now to get things working. Besides the e2e not working, TypeORM threw a bunch of errors about `smart_search` not existing.